File: C#JAVA_Client_Server/ClientServcerC#Protocols/Solution1/Repository/CursaDBRepository.cs
using System;
using System.Collections.Generic;
using Domain;
using Domain.validators.interfaceAndException;
using MySql.Data.MySqlClient;

namespace Repository
{
    public class CursaDBRepository : IObjectRepo<Cursa,int>
    {
        private IValidator<Cursa> _validator;

        private MySqlDataAdapter dataAdapter=new MySqlDataAdapter();
        private Utils.Utils _utils;

        public CursaDBRepository(IValidator<Cursa> validator)
        {
            _validator = validator;
            _utils = new Utils.Utils();
        }

        public Cursa FindOne(int id)
        {
            var connection = _utils.GetConnection();
            var commander = _utils.GetCommander(connection);
            try
            {
                connection.Open();
                var sql = "select * from cursa where idCursa=@idCursa";
                commander = new MySqlCommand(sql, connection);
                commander.Parameters.AddWithValue("@idCursa", id);
                MySqlDataReader reader = commander.ExecuteReader();
                Cursa cursa = new Cursa();
                reader.Read();
                cursa.NumeCupa = reader.GetString(2);
                cursa.CapacitateMotor = reader.GetInt32(1);
                cursa.Id = reader.GetInt32(0);
                return cursa;
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }

        public MySqlDataAdapter getDataAdapter()
        {
            dataAdapter.SelectCommand = new MySqlCommand("select * from cursa", _utils.GetConnection());
            return dataAdapter;
        }
        public IEnumerable<Cursa> FindAll()
        {
            var connection = _utils.GetConnection();
            var commander = _utils.GetCommander(connection);
            try
            {
                List<Cursa> curse = new List<Cursa>();
                
                connection.Open();
                var sql = "select * from cursa";
                commander = new MySqlCommand(sql, connection);
                MySqlDataReader reader = commander.ExecuteReader();
                while (reader.Read())
                {
                    Cursa cursa = new Cursa(reader.GetString(2), reader.GetInt32(1));
                    cursa.Id = reader.GetInt32(0);
                    curse.Add(cursa);
                }

                return curse;
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }

        public void Save(Cursa entity)
        {
            try
            {
                _validator.validate(entity);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                return;
            }
            var connection = _utils.GetConnection();
            var commander = _utils.GetCommander(connection);
            try
            {
                connection.Open();
                var sql = "Insert into cursa(capacitateMotor,numeCupa) values(@capacitate ,@numeCupa)";
                commander = new MySqlCommand(sql
                    , connection);
                commander.Parameters.AddWithValue("@capacitate", entity.CapacitateMotor);
                commander.Parameters.AddWithValue("@numeCupa", entity.CapacitateMotor);
                commander.Prepare();

                commander.ExecuteNonQuery();
                
                Console.WriteLine("Row Inserted");
                connection.Close();

            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }

        public void Update(int id,Cursa cursa)
        {
            try
            {
                _validator.validate(cursa);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                return;
            }
            var connection = _utils.GetConnection();
            var commander = _utils.GetCommander(connection);
            try
            {
                connection.Open();
                var sql = "update cursa set capacitateMotr=@capacitateMotor where idCursa=@idCursa";
                commander = new MySqlCommand(sql
                    , connection);
                commander.Parameters.AddWithValue("@capacitateMotor", 2000);
                commander.Parameters.AddWithValue("@idCursa", id);
                commander.Prepare();

                commander.ExecuteNonQuery();
                
                Console.WriteLine("Row updated");
                connection.Close();

            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }        }

        public void Delete(int id)
        {
            var connection =_utils.GetConnection();
            var commander =_utils.GetCommander(connection);
            try
            {
                connection.Open();
                var sql = "delete from cursa where idCursa=@idCursa";
                commander = new MySqlCommand(sql
                    , connection);
                commander.Parameters.AddWithValue("@idCursa", id);
                commander.Prepare();

                commander.ExecuteNonQuery();
                
                Console.WriteLine("Row deleted");
                connection.Close();

            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }        
        }
    }
}

File: C#JAVA_Client_Server/ClientServcerC#Protocols/LongExample/Repo/src/main/java/repository/database/CursaDBRepository.java
package repository.database;

import entities.domain.Cursa;
import entities.domain.validator.interfaceAndExceptions.Validator;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import repository.CursaRepository;
import repository.JdbcUtils;


import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class CursaDBRepository implements CursaRepository {
    private Validator<Cursa> validator;

    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger(CursaDBRepository.class);

    public CursaDBRepository(Properties props, Validator<Cursa> validator) {
        this.validator=validator;
        logger.info("Initializing CarsDBRepository with properties: {} ",props);
        this.dbUtils =new JdbcUtils(props);    }

    @Override
    public Cursa findOne(Integer integer) {
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from cursa where idCursa=?"))
        {
            preStmt.setInt(1,integer);
            try(ResultSet resultSet=preStmt.executeQuery()){
                resultSet.next();
                int id=resultSet.getInt("idCursa");
                int capacitateMotor=resultSet.getInt("capacitateMotor");
                String numeCupa=resultSet.getString("numeCupa");
                Cursa cursa=new Cursa(numeCupa,capacitateMotor);
                cursa.setId(id);
                return cursa;
            }
        }catch (SQLException e){
            logger.error(e);
            System.err.println("Error DB"+e);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit(integer);
        return null;
    }

    @Override
    public Iterable<Cursa> findAll(){
        logger.traceEntry();
        Connection con= dbUtils.getConnection();
        List<Cursa> curse=new ArrayList<>();
        try(PreparedStatement preStmt=con.prepareStatement("select * from cursa"))
        {
            try(ResultSet resultSet=preStmt.executeQuery()){
                while (resultSet.next()){
                    int id=resultSet.getInt("idCursa");
                    Integer capacitateMotor=resultSet.getInt("capacitateMotor");
                    String numeCupa=resultSet.getString("numeCupa");
                    Cursa cursa=new Cursa(numeCupa,capacitateMotor);
                    cursa.setId(id);
                    curse.add(cursa);
                }
            }
        }catch (SQLException e){
            logger.error(e);
            System.err.println("Error DB"+e);
        }
        logger.traceExit(curse);
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        return curse;
    }

    @Override
    public void save(Cursa cursa) {
        logger.traceEntry("saving task {}",cursa);
        try{
            validator.validate(cursa);
        }catch (Exception e){
            logger.error(e);
            System.err.println("ERROR DOMAIN"+e);
            logger.traceExit();
            return;
        }
        Connection con= dbUtils.getConnection();
        try (PreparedStatement preparedStatement=con.prepareStatement("insert into cursa (capacitateMotor,numeCupa)values (?,?)")){
            preparedStatement.setInt(1,cursa.getCapacitateMotor());
            preparedStatement.setString(2,cursa.getNumeCupa());
            int result=preparedStatement.executeUpdate();
            logger.trace("Saved {} instance",result);
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit();
    }

    @Override
    public void update(Integer integer,Cursa cursa) {
        logger.traceEntry("update task {}",integer);
        try{
            validator.validate(cursa);
        }catch (Exception e){
            logger.error(e);
            System.err.println("ERROR DOMAIN"+e);
            logger.traceExit();
            return;
        }
        Connection con= dbUtils.getConnection();
        try (PreparedStatement preparedStatement=con.prepareStatement("update cursa set capacitateMotr=? where idCursa=?")){
            preparedStatement.setString(3,cursa.getNumeCupa());
            preparedStatement.setInt(2,cursa.getCapacitateMotor());
            preparedStatement.setInt(1,integer);
            int result=preparedStatement.executeUpdate();
            logger.trace("updated {} instance",result);
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Integer integer) {
        logger.traceEntry("delete task {}",integer);
        Connection con= dbUtils.getConnection();
        try (PreparedStatement preparedStatement=con.prepareStatement("delete from cursa where idCursa=?")){
            preparedStatement.setInt(1,integer);
            int result=preparedStatement.executeUpdate();
            logger.trace("deleted {} instance",result);
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        try {
            con.close();
        } catch (SQLException throwables) {
            logger.error(throwables);
            System.err.println("ERROR DB"+throwables);
        }
        logger.traceExit();
    }
}
